Add a party overview page

me-and opened the pull request:
I've added a party overview page to the data display tool.

At the moment it's just a prototype – it has the information but none of the nice formatting yet. Before I go and make it pretty though, I want to check you're happy with the broad concept, the information that's included (and not included), the layout, etc.

The plan is that the page will use table formatting similar to the task overview page, but other than formatting I don't think there's anything I can immediately think of to change, so this seemed like an ideal time to get some commentary.

There's [a live version](https://tastycake.net/~adam/nothingtoseehere.html) if that's easier to play with; the only differences are that it's based on my [party-overview-unofficial](https://github.com/me-and/tools-for-habitrpg/tree/party-overview-unofficial) branch, which also contains the changes to add optionally storing login details and to make it clear that it's not at all official.

me-and committed:
Add placeholder for Party Overview page

me-and committed:
Add raw user data to party overview

me-and committed:
Very rough and ready party overview

Use a basic HTML table to build a party overview displaying usernames
and HP.

me-and committed:
Add lots more data to party overview

Alys commented:
Oh my gosh! This is lovely!! I definitely want this!

Yes, please do go ahead and pretty it up. We will certainly use it.

The only suggestion I have at the moment is that the Questing column should distinguish between a player who has rejected the quest and one who has not yet responded.

me-and commented:
Good call on the questing column; I'd only looked at the output my party is currently generating (we're on a quest right now) and didn't think to consider handling other scenarios ><

me-and committed:
Handle parties not on a quest

Handle the case where parties are not on a quest by spotting when the
quest data is null (rather than just erroring out) and skipping the
quest information column in the party overview in that scenario.

Also re-order the columns on the party overview page so the text columns
(user, class, questing status) all appear to the left of the numerical
columns.

me-and committed:
WIP
